src/components/Dialog/DialogForm.tsx

```diff
@@ -119,17 +123,23 @@ export function DialogForm<T, R = void>({
       decorators={defaultDecorators}
       {...FormProps}
       onSubmit={async (data, form) => {
-        if (onlyDirtySubmit && !form.getState().dirty) {
+        const submitAction = (data as any).submitAction;
+        if (
+          (!sendIfClean && !form.getState().dirty) ||
+          (typeof sendIfClean === 'string' && sendIfClean !== submitAction)
+        ) {
           onClose?.('cleanSubmit', form);
           return null;
         }
-        try {
-          const res = await onSubmit(data, form);
-          onSuccess?.(res);
-          onClose?.('success', form);
-          return null;
-        } catch (e) {
-          return await handleFormError(e, errorHandlers);
+        if (sendIfClean || sendIfClean === submitAction) {
```

Comment:
Is this if statement catching anything different than the else case of the if statement above?

Reply:
It should have been `if (sendIfClean === true || sendIfClean === submitAction)`

`sendIfClean` could have a value other than `true` but that doesn't match `submitAction`; that's what I was trying to catch

Reply:
(It's fixed now)

Reply:
Actually, this just needs to be reworked a bit....

Reply:
Okay, derpderpderp, the real problem is **Ryan can't read** and didn't see the second line in the first if statement:

`(typeof sendIfClean === 'string' && sendIfClean !== submitAction)`

So you're right, @Carson, that second if statement was useless and has now been removed.

Reply:
😆  Cheers 🍻